src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -970,10 +989,18 @@ private bool ImportIntrinsicCall(MethodDesc method)
             return false;
         }
 
-        private void HandleCall(MethodDesc callee, ILOpcode opcode = ILOpcode.call)
-        { 
-            AddMethodReference(callee);
-            int offset = GetTotalParameterOffset() + GetTotalLocalOffset() + callee.Signature.ReturnType.GetElementSize().AsInt;
+        private void HandleCall(MethodDesc callee, MethodSignature signature, ILOpcode opcode = ILOpcode.call, LLVMValueRef calliTarget = default(LLVMValueRef))
+        {
+            if (opcode == ILOpcode.callvirt && callee.IsVirtual)
+            {
+                AddVirtualMethodReference(callee);
+            }
+            else if (callee != null)
+            {
+                AddMethodReference(callee);
+            }
+
+            int offset = GetTotalParameterOffset() + GetTotalLocalOffset() + signature.ReturnType.GetElementSize().AsInt;
```

Comment:
Is this doing the right thing for `void` return types?

Reply:
We were probably allocating space for void return types and definitely emitting address calculation code. Changed to emit nulls for void methods.